pandas/_libs/lib.pyx

```diff
@@ -2359,3 +2356,103 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline void convert_and_set_item(object item, Py_ssize_t index,
+                                      object[:] result,
+                                      bint keep_trivial_numbers):
+    cdef:
+        bint do_convert = 1
+        double double_item
```

Comment:
we try to use `float64_t` instead of `double` unless there is a compelling reason

Reply:
Can change, though the difference eludes me for now, could you please enlighten me? :)

Reply:
We adopted this convention exactly because the difference is really obscure (I'm pretty sure it was explained to me somewhere on the cython GH tracker).  In most cases they are the same, but `double` is only "assured" to be twice as precise as `single`, which esoteric platforms can make their own decisions about.

Uh, don't quote me on this.  Me-being-confused is a big part of why this was changed in the first place.

Reply:
Changed to float64_t.